mlcommons/medperf
# CLI MLCube Timeout

This PR adds functionality for specifying timeouts for each task independently. Closes https://github.com/mlcommons/medperf/issues/140.

* Implement rest tests

* Removed unused dependency

* Simplify text setter

* fix hidden input not returning anything

* Implement cli ui tests

* Remove duplicated authenticate method

* remove unnecessary pass statements

* REST follows comms init signature

* Fix Associate command

* Provide more user-friendly error on Dataset init

* Replace exceptions to pretty errors

* Remove UI circular import

* Test new pretty errors

* Fix comment typo

* Implement associate tests

* Move prompt logic to comms login method

* Fix test with login method

* Add missing abstract method hidden prompt

* Implement login command tests

* Fix login method

* Implement additional tests

* split execution into methods

* reorder methods

* Add cleanup after benchmark execution

* fix comms not receiving ui instance

* Fix registration not using ui for prompt

* Refactor prepare command into methods

* Add cleanup to preparation flow

* Store datasets under their generated_uid

* Split dset preparation and registration commands

* Adjust tests to dataset changes

* Initial submit command

* Fix Benchmark Execution not finding results path

* DRY up comms and ui fixtures

* Remove unused deprecated method

* Implement prepare command tests

* start working on datasets command tests

* Test preparation flow

* Implement benchmark execution tests

* Fix results_path not returning anything

* add results filename to config

* Implement missing benchmark exec tests

* Implement register tests

* Move shared function to utils

* Implement submit tests

* Fix execute command data uid as str

* Fix pretty error not receiving ui

* Remove leftover files from merge

* Recover comms and ui fixtures

* Fix broken comms fixture

* Remove deprecated file

* Fix result not using dataset server uid

* Fix ensure storage is setup before anything

* Fix prepare not storing/registering datset

* Fix detection of already prepared datasets

* Add docstring to submit command

* Fix login test not mocking "exist" method

* Implement draft for test command

* Add method to create local cubes

* Split init logic from preparation logic

* [WIP] Develop methods for test command

* Store config variables as module variables

* Convert config references to new structure

* Convert config references to new structure

* Convert outdated config references to new struct

* Add helper function for storage relative paths

* Fix storage path parsing

* Fix credentials_path not relative to storage

* Formatting

* Fix outdated references to config

* Undo config changes

* Add initial testing implementation

* Add run_test parameter to benchmark exec

* Fix resolve argument paths instead of forcing abs

* Implement cube caching

* Add local cube prefix to config

* Add ui object to cube get calls

* Add logging to set_registration

* Add run_test to prepare step

* Move cleanup outside of function declaration

* Fix decorators disorder

* Add testing prefixes

* Remove unused print

* Add uid to dataset

* Ensure comparison is done with same datatype

* Remove unnecessary print

* Only check mlcube path vailidity if provided

* Return ids after finishing the test

* Add logging messages to test flow

* Change name to evade pytest issues

* Fix outdated test execution imports

* Reorder init args. Fix model validation

* Remove unused import

* Implement initial tests

* Fix dataset using incomplete path

* Move comms and ui to config singleton

* Remove unused argument for registration write

* Fix medperf logs not working correctly

* Fix results not being stored in correct path

* Fix results_path_testing

* Run tests on test storage path

* Implement mlcube list method

* Add tests for mlcube list

* Implement mlcube submit method

* Implement mlcube associate command

* Implement submit tests

* Implement associate tests

* Implement REST additional tests

* Add new dependencies to medperf cli

* Implement benchmark list subcommand

* Implement get_user_benchmarks test

* Implement benchmark submit subcommand

* Implement list all benchmarks

* Implement list all mlcubes

* Update CLI readme

* Update outdate config and import calls

* Add demo_dataset related info to test benchmark

* Fix typo

* Add benchmarks directory to medperf storage fs

* Implement local benchmark storage

* restructure is_valid

* Display approved status

* Add demo dataset to benchmark submission

* Fix dset list bugs. Add Server UID to list

* Benchmark Association initial implementation

* Add ui messages on successful association request

* Add associate command

* Fix outdated import

* Download demo dataset for test. Disable validation

* Implement testing for all cubes in benchmark

* Add compatibility test to bmk submit

* Update seed benchmark

* Fix test command evaluator arg

* Remove demo dataset prefix from benchmark

* Add mlcube association to benchmark

* Fix typo in parameters url

* Fix not being able to submit cubes without tarball

* Fix results entity tests

* Move compatibily test unit tests

* Fix utils test errors

* Fix compatibility test unit tests

* Fix submit errors

* Fix dataset create tests

* Fix tests for results_path

* Fix result create tests

* Fix result entity tests

* Fix dataset entity tests

* Write additional tests

* Add tests to benchmark object

* Implement additional comp_test tests

* Add download_demo_data tests

* Rename dset assoc test

* Add missing tests to benchmark assoc

* Add additional tests to submit

* Fix pytest not stunting listdir

* Add return description to docstring

* Fix bmk-dset association

* Fix list typing error

* Move comptest tests to correct location

* Move comptests to correct location

* Update comptest references

* Update all imports to new structure

* rename ui and comms interfaces

* Run Comptest on dataset assoc

* Fix tarfile creating superfluous files

* Fix compatibility tests

* Fix: use correct uid for result creation

* Fix missing ui error

* Add compatibility tests to associations

* Pass result tests through associations

* Fix test error

* Implement list associations command

* Fix associations

* Add associations tests

* Imlement rest tests

* Implement approval commands

* Move associations to list command

* Rename list to ls

* Implement approval tests

* Fix dataset assoc approval implementation

* Add missing tests

* remove unused cache command

* Store demo datasets in a persistent location

* Fix run comp test with local demo dataset

* Allow running tests with demo dataset

* Fixed ui and comms imports

* Remove python3.9 dependency

* Add missing self.ui

* Fix remove tmp_submission on cleanup

* Fix return user benchmarks

* Add tests for user benchmarks returned

* Fix comp test failing for demo dset already prep

* Add init files to command tests

* Fix registration tests

* Fix bug set approval status always approves

* Update decorator order of execution

* Cli remove init files (#89)

* Update all imports to new structure

* rename ui and comms interfaces

* Fix decorator imports

* Fix symlink failing

* Force benchmark update when not testing

* Remove redefinition of Benchmark

* Fix formatting errors

* Fix formatting

* Cli offline testing (#128)

* Authenticate only when needed

* Fix outdated auth_req test

* Add optional timeouts to all mlcube executions

* Fix/Add timeout testing

* Reformat code
